src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -499,60 +499,172 @@ private string JoinPackageNames(IEnumerable<Package> pkgs, string seperator = ",
+                        var message = string.Format(Resources.MessageSamePackageDiffVersInBuiltinPackages, packageToDownload, dupPkg,
+                            DynamoViewModel.BrandingResourceProvider.ProductName);
+
+                        MessageBoxService.Show(message, Resources.CannotDownloadPackageMessageBoxTitle,
+                            MessageBoxButton.OK, MessageBoxImage.Exclamation);
+                    }
+                    return false;// All conflicts with built-in packages must be first resolved manually before continuing to download.
+                }
+
+                if (package.version == duplicatePackage.VersionName)
+                {
+                    var message = string.Format(Resources.MessageSamePackageSameVersInLocalPackages, packageToDownload);
                     MessageBoxService.Show(message, Resources.CannotDownloadPackageMessageBoxTitle,
-                        MessageBoxButton.OK, MessageBoxImage.Information);
+                    MessageBoxButton.OK, MessageBoxImage.Information);
+
+                    return false;
                 }
                 else
                 {
-                    var message = string.Format(Resources.MessageSamePackageDiffVersInBuiltinPackages,
-                        packageToDownload, JoinPackageNames(new[] { duplicatePackage }),
-                        DynamoViewModel.BrandingResourceProvider.ProductName);
+                    MessageBoxResult dialogResult;
 
-                    MessageBoxService.Show(message, Resources.CannotDownloadPackageMessageBoxTitle,
-                        MessageBoxButton.OK, MessageBoxImage.Exclamation);
+                    if (duplicatePackage.InUse(DynamoViewModel.Model))
+                    {// Loaded assemblies or pacakge in use in workspace
+                        dialogResult = MessageBoxService.Show(string.Format(Resources.MessageSamePackageDiffVersInLocalPackages, packageToDownload, dupPkg),
+                            Resources.LoadedPackagesConflictMessageBoxTitle,
+                            MessageBoxButton.OKCancel, new string[] { Resources.UninstallLoadedPackage, Resources.GenericTaskDialogOptionCancel }, MessageBoxImage.Exclamation);
+
+                        if (dialogResult == MessageBoxResult.OK)
+                        {
+                            // mark for uninstallation
+                            duplicatePackage.MarkForUninstall(DynamoViewModel.Model.PreferenceSettings);
+                        }
+                        return false;// Any package that is in use must first be uninstalled before continuing to download.
+                    }
+                    else
+                    {
+                        dialogResult = MessageBoxService.Show(String.Format(Resources.MessageAlreadyInstallDynamo,
+                            DynamoViewModel.BrandingResourceProvider.ProductName, dupPkg, packageToDownload), 
+                            Resources.CannotDownloadPackageMessageBoxTitle,
```

Comment:
Does `CannotDownloadPackageMessageBoxTitle` title still make sense now as the package _**can**_ now be installed right?

Reply:
replaced with `PackagesInUseConflictMessageBoxTitle` 
```
Package has conflicts with one or more packages in use.
```